delivery_procurement_group_carrier/models/stock_picking.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _align_group_carrier(self):
+        for picking in self:
+            picking_carrier = picking.carrier_id
+            picking_group = picking.group_id
```

Comment:
You should check if there is a group set, if not ignore it. 
```suggestion
            picking_group = picking.group_id
            if not picking_group:
                continue
```

Reply:
done